src/plugins/bearer/connman/qconnmanservice_linux.cpp

```diff
@@ -504,7 +504,12 @@ void QConnmanTechnologyInterface::scanReply(QDBusPendingCallWatcher *call)
 {
     QDBusPendingReply<void> props_reply = *call;
     if (props_reply.isError()) {
-        qCWarning(qLcLibBearer) << "QConnmanTechnologyInterface::scanReply()" << props_reply.error().message();
```

Comment:
So this error eventually comes from [ConnMan technology.c:scan()](https://github.com/sailfishos/connman/blob/master/connman/src/technology.c#L1281) that is called via [libconnman-qt Scan()](https://github.com/sailfishos/libconnman-qt/blob/master/libconnman-qt/networktechnology.cpp#L373) ? I wonder if the actual problem lies somewhere deep, it'd be interesting to check what ConnMan does actually return in case the "NoCarrier" is set as an error.

Reply:
Ah yeah, this happens when the WiFi is not simply powered. So `ENOLINK` is returned when attempting to do a scan. I guess what ConnMan does is ok but higher level components should refrain from scanning when they see that device is not powered.

Reply:
So like
```
void QConnmanTechnologyInterface::scan()
{
    if (!getProperty(QStringLiteral("Powered")).toBool()) {
        qCDebug(qLcLibBearer) << "QConnmanTechnologyInterface::scan() technology not powered";
        return;
    }
```
? I cannot test as for some reason this does not compile on my SDK now, that is because of other unknown reasons.

Reply:
Will give a try tomorrow.